src/vs/workbench/browser/positronDataExplorer/components/dataExplorerPanel/components/addEditRowFilterModalPopup/components/columnSelectorDataGridInstance.tsx

```diff
@@ -47,24 +42,56 @@ export class ColumnSelectorDataGridInstance extends DataGridInstance {
 
 	//#endregion Private Properties
 
+	//#region Static Methods
+
+	/**
+	 * Creates a new column selector data grid instance.
+	 * @param dataExplorerClientInstance The data explorer client instance.
+	 * @returns A Promise<ColumnSelectorDataGridInstance> that resolves when the operation is
+	 * complete.
+	 */
+	public static async create(
+		dataExplorerClientInstance: DataExplorerClientInstance,
+	): Promise<ColumnSelectorDataGridInstance | undefined> {
+		try {
+			// Get the backend state so that we can get the initial number of columns.
+			const backedState = await dataExplorerClientInstance.getBackendState();
+
+			// Return a new instance of the column selector data grid instance.
+			return new ColumnSelectorDataGridInstance(
+				backedState.table_shape.num_columns,
+				dataExplorerClientInstance
+			);
+		} catch {
+			return undefined;
```

Comment:
If not propagating the exception here, we should log it before discarding.

Reply:
I've addressed this issue using a modal dialog:

<img width="1530" alt="image" src="https://github.com/user-attachments/assets/23ffc15f-6233-4584-a082-895fcc427d1c">

Reply:
Using the notification service wasn't a great idea in this case because the user is already in a modal popup, which causes notifications to be dimmed below the modal popup overlay.